tests/forms_tests/tests/test_forms.py

```diff
@@ -3576,3 +3576,17 @@ class DataForm(Form):
         f = DataForm({'data': 'xyzzy'})
         self.assertTrue(f.is_valid())
         self.assertEqual(f.cleaned_data, {'data': 'xyzzy'})
+
+    def test_choicefield_uses_separate_widget_instances(self):
```

Comment:
We don't have strict guidance about this, but I think the test you've written is sufficient. Some `Field` tests will inevitably require more specific classes.